ipaserver/install/dogtaginstance.py

```diff
@@ -332,13 +332,33 @@ def secure_ajp_connector(self):
         if self.__is_newer_tomcat_version('9.0.31.0'):
             secretattr = 'secret'
 
-        rewrite = True
-        if secretattr in connector.attrib:
-            # secret is already in place
-            # Perhaps, we need to synchronize it with Apache configuration
-            self.ajp_secret = connector.attrib[secretattr]
-            rewrite = False
-        else:
+        # AJP protocol is at version 1.3. With IPv4/IPv6 split, there might
+        # be multiple AJP adapters; update them all.
+        #
+        # First, iterate through all adapters and see if any of them have a
+        # secret value set.
+        for connector in connectors:
+            if secretattr in connector.attrib or oldattr in connector.attrib:
+                # secret is already in place
+                #
+                # Perhaps, we need to synchronize it with Apache configuration
+                # or other AJP connector entries. Save it so we know we've
+                # found at least one. Because in our next loop we update the
+                # config value if incorrect, it is safe to overwrite
+                # self.ajp_adapter -- in the worst case, we'll create an
+                # entirely new value if this element happened to have an
+                # empty secret value. Plus, IPA is in charge of managing the
+                # value for the httpd side of the AJP connection as well.
+                self.ajp_secret = connector.attrib[secretattr]
```

Comment:
For test case 1a where requiredSecret="testing_ajp_secret". Because __is_newer() is True then secretattr  = 'secret'. The conditional on line 341 looks for both the old and new attr in connector.attrib. Since requiredSecret is there this is satisfied but since 'secret' is not there this blows on up line 352 with a KeyError.

Reply:
Ah we probably want an `if`/`elseif` here instead, gotcha.